test/c/test_circuit.c

```
@@ -254,6 +263,40 @@ int test_gate_num_params(void) {
     return Ok;
 }
 
+/**
+ * Test edge cases for getting the op counts.
+ */
+int test_get_gate_counts(void) {
+    QkCircuit *qc = qk_circuit_new(3, 3);
+
+    // test empty circuit
+    int result = Ok;
+    QkOpCounts c1 = qk_circuit_count_ops(qc);
+    if (c1.len != 0) {
+        result = EqualityError;
```

Why not failing fast and "goto"ing to a cleanup label?

---

`goto` statements are banned in my vocabulary, I need to remember that sometimes they are ok 😉 added!